earn/src/components/boost/ImportBoostWidget.tsx

```diff
@@ -394,26 +401,68 @@ export default function ImportBoostWidget(props: ImportBoostWidgetProps) {
     enabled: enableHooks && Boolean(userAddress),
   });
 
+  // The NFT indices we can use if the user has some unused BorrowerNFTs
+  useEffect(() => {
+    let mounted = true;
+
+    (async () => {
+      if (!userAddress) return;
+      const chainId = (await provider.getNetwork()).chainId;
+      const results = await fetchListOfBorrowerNfts(chainId, provider, userAddress, {
+        validUniswapPool: cardInfo.uniswapPool,
+        onlyCheckMostRecentModify: true,
+        includeFreshBorrowers: true,
+      });
+
+      if (mounted && results.borrowers.length > 0) {
+        setAvailableNft({ borrower: results.borrowers[0], ptrIdx: results.indices[0] });
+      }
+    })();
+
+    return () => {
+      mounted = false;
+    };
+  }, [provider, userAddress, cardInfo, setAvailableNft]);
+
+  // If we're reusing an old NFT, check whether it has ANTE
+  const { data: borrowerBalance } = useBalance({
+    address: availableNft?.borrower ?? '0x',
+    chainId: activeChain.id,
+    watch: false,
+    enabled: availableNft !== undefined,
+  });
+
+  const ethToSend = useMemo(() => {
+    if (!marketParameters) {
+      return ethers.BigNumber.from(0);
+    }
+    if (availableNft !== undefined && borrowerBalance !== undefined) {
+      if (borrowerBalance.value.lt(marketParameters.ante)) return marketParameters.ante.sub(borrowerBalance.value);
+      return ethers.BigNumber.from(0);
+    }
+    return marketParameters.ante;
+  }, [marketParameters, availableNft, borrowerBalance]);
+
   // Prepare for actual import/mint transaction
   const borrowerNft = useMemo(() => new ethers.utils.Interface(borrowerNftAbi), []);
   // First, we `mint` so that they have a `Borrower` to put stuff in
   const encodedMint = useMemo(() => {
-    if (!userAddress) return undefined;
+    if (!userAddress) return '0x';
```

Comment:
Why change?

Reply:
To get rid of the annoying ternary in the usePrepareContractWrite call. Because now we have _another_ ternary there, so if we left this as returning undefined it would be nested ternaries.